server/channels/store/searchlayer/file_info_layer.go

```diff
@@ -124,14 +127,36 @@ func (s SearchFileInfoStore) AttachToPost(rctx request.CTX, fileId, postId, chan
 	return err
 }
 
-func (s SearchFileInfoStore) DeleteForPost(rctx request.CTX, postId string) (string, error) {
-	result, err := s.FileInfoStore.DeleteForPost(rctx, postId)
+func (s SearchFileInfoStore) DeleteForPost(rctx request.CTX, postID string) (string, error) {
+	// temporary workaround because deleteFileIndexForPost is not working due to the post_id not being indexed with the file
```

Comment:
Did `deleteFileIndexForPost` ever work? I don't understand if the change is needed because this PR changes things or if `deleteFileIndexForPost` has been broken before.

Reply:
I don't think deleteFileIndexForPost has ever worked. I filed this ticket for it https://mattermost.atlassian.net/browse/MM-59264